src/components/inputOfTodos.tsx

```diff
@@ -0,0 +1,75 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React from 'react';
+import { Todo } from '../types/Todo';
+
+interface Props {
+  setNewTodo: (newTodo: string) => void;
+  newTodo: string;
+  addTodo: (title: string) => void;
+  todos: Todo[];
+  setTodos: (todos: Todo[]) => void;
+}
+
+export const InputOfTodos: React.FC<Props> = ({
+  setNewTodo, newTodo, addTodo, todos, setTodos,
+}) => {
+  const handleAddNewTodo = () => {
+    if (newTodo) {
+      addTodo(newTodo);
+      setNewTodo('');
+    }
+  };
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTodo(event.target.value);
+  };
+
+  const handleInputKeyPress
+  = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Enter') {
+      event.preventDefault();
+      handleAddNewTodo();
+    }
+  };
+
+  const handleTaggleAll = () => {
+    let updatedTodos = todos;
+
+    if (todos.some((t) => !t.completed)) {
+      updatedTodos = todos.map((t) => ({
+        ...t,
+        completed: true,
+      }));
+    } else {
+      updatedTodos = todos.map((t) => ({
+        ...t,
+        completed: !t.completed,
+      }));
+    }
+
+    setTodos(updatedTodos);
+  };
+
+  return (
+    <header className="todoapp__header">
+      {todos.length > 0 && (
+        <button
+          type="button"
+          className={`todoapp__toggle-all ${todos.every((t) => t.completed) ? 'active' : ''}`}
```

Comment:
You itarate over the array of Todos in two places to get the same information (see line 38). This is not the best performance optimization. Let's try to do this only once and use the result in both places.